src/Microsoft.SqlTools.ServiceLayer/LanguageServices/AutoCompleteHelper.cs

```diff
@@ -680,5 +680,79 @@ private static string GetCompletionItemInsertName(Declaration autoCompleteItem)
                 return null;
             }
         }
+
+        /// <summary>
+        /// Converts a SQL Parser List of MethodHelpText objects into a VS Code SignatureHelp object
+        /// </summary>
+        internal static SignatureHelp ConvertMethodHelpTextListToSignatureHelp(List<Babel.MethodHelpText> methods, Babel.MethodNameAndParamLocations locations, int line, int column)
+        {
+            SignatureHelp help = new SignatureHelp();
+
+            help.Signatures = methods.Select(method =>
+            {
+                return new SignatureInformation()
+                {
+                    // Signature label format: <name> param1, param2, ..., paramn RETURNS <type>
+                    Label = method.Name + " " + method.Parameters.Select(parameter => parameter.Display).Aggregate((l, r) => l + "," + r) + " " + method.Type,
+                    Documentation = method.Description,
+                    Parameters = method.Parameters.Select(parameter => 
+                    {
+                        return new ParameterInformation()
+                        {
+                            Label = parameter.Display,
+                            Documentation = parameter.Description
+                        };
+                    }).ToArray()
+                };
+            }).ToArray();
+
+            if (help.Signatures.Length == 0)
+            {
+                return null;
+            }
+
+            // Find the matching method signature at the cursor's location
+            int signatureIndex = methods.FindIndex(method => method.Name.Contains(locations.Name));
```

Comment:
I agree, this makes sense. I'm not sure why the protocol dictates returning all signatures (perhaps to list multiple signatures if there are overloads? (not sure if this applies to SQL)). OmniSharp appears to follow this pattern:  https://github.com/OmniSharp/omnisharp-roslyn/blob/cdd99aef194660835e61c58f19fb51ab3dabc52e/src/OmniSharp.Roslyn.CSharp/Services/Signatures/SignatureHelpService.cs#L24